pkg/provider/ibm/provider.go

```diff
@@ -575,16 +589,51 @@ func (ibm *providerIBM) NewClient(ctx context.Context, store esv1beta1.GenericSt
 		storeKind: store.GetObjectKind().GroupVersionKind().Kind,
 	}
 
-	if err := iStore.setAuth(ctx); err != nil {
-		return nil, err
-	}
+	var err error
+	var secretsManager *sm.SecretsManagerV1
+	secretContainerAuthProfile := iStore.store.Auth.SecretRef.SecretContainerAuth.Profile
+	if secretContainerAuthProfile != "" {
+		// container-based auth
+		secretContainerAuthToken := iStore.store.Auth.SecretRef.SecretContainerAuth.TokenLocation
+		secretContainerAuthEndpoint := iStore.store.Auth.SecretRef.SecretContainerAuth.IAMEndpoint
+
+		if secretContainerAuthToken == "" {
+			// API default path
+			secretContainerAuthToken = "/var/run/secrets/tokens/vault-token"
+		}
+		if secretContainerAuthEndpoint == "" {
+			// API default path
+			secretContainerAuthToken = "https://iam.cloud.ibm.com"
+		}
 
-	secretsManager, err := sm.NewSecretsManagerV1(&sm.SecretsManagerV1Options{
-		URL: *storeSpec.Provider.IBM.ServiceURL,
-		Authenticator: &core.IamAuthenticator{
-			ApiKey: string(iStore.credentials),
-		},
-	})
+		authenticator, err := core.NewContainerAuthenticatorBuilder().
+			SetIAMProfileName(secretContainerAuthProfile).
+			SetCRTokenFilename(secretContainerAuthToken).
```

Comment:
Is there an opt to specify the token as string? I suggest reading the service account token via API instead of reading it from file. 
Then we dont have to rely on volumeMounts from the controller

Reply:
Unfortunately no, this is contrary to the model of container-based authentication.  The IAM token has a maximum lifetime of one hour in the IBM environment.  Instead of explicitly granting permission based on a key, it is implicitly granted using other attributes the API server can verify as part of the trusted profile being used (namespace of operator, cluster ID, data center location, etc.)  There is more up-front setup time to avoid needing to manage security, creation, rotation, etc. of the "secret-zero" - the trade-off is more security and less ongoing maintenance of the external-secrets access/authorization.

There might be a set of specific curl or other API calls that could retrieve the container resource token and get the corresponding service account token without needing a volume mount, but this isn't mentioned anywhere in the documentation, so I think it would be best to stick with the recommended flow for this feature.

Nice summary: https://www.ibm.com/cloud/blog/introducing-trusted-profiles-for-vpc-virtual-server-instances

Further technical references:

* https://github.com/IBM/go-sdk-core/blob/main/Authentication.md#container-authentication
* https://cloud.ibm.com/docs/containers?topic=containers-pod-iam-identity&interface=ui
* https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection

Reply:
Other cloud providers have similar technology to provide a identity via service accounts: IRSA (AWS) and Workload Identity (GCP, Azure). We managed to integrate that in a way that allows us use other Kubernetes `Kind=ServiceAccount` for authenticating with the cloud provider - NOT just the service account that is mounted inside the controller. This allows users to manage the lifecycle of the ServiceAccount independently from the operator (e.g. it is operated by a different team).
Under the hood we issue short-lived jwt tokens (via TokenRequest API) with the correct set of claims which are then used for authentication. The very same thing that IRSA & WorkloadIdentity webhook does.

Looking up the token via file is fine for me and a big step in the right direction, i just want to raise awareness that this is a thing we should support.

(i'm unresolving this conversation just to make sure it doesn't get lost, there's no need to do anything in that regard)

Reply:
Per later discussion, it sounds like this specific concern no longer has a future path forward as the IBM token is automatically refreshed in the cluster API layer, which is why it requires the volume mount.